assets/js/modules/subscribe-with-google/components/plugins/AccessSelector.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * Subscribe with Google Account Create component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_NAME } from '../../datastore/constants';
+
+const { compose } = global.wp.compose;
+const { withDispatch, withSelect } = global.wp.data;
+
+export class AccessSelector extends Component {
+	render() {
+		if ( ! this.props.products ) {
+			return null;
+		}
+
+		const { SelectControl, PanelRow } = global.wp.components;
+		const { PluginDocumentSettingPanel } = global.wp.editPost;
+
+		// Only show for normal posts.
+		const { select } = global.wp.data;
+		const postType = select( 'core/editor' ).getCurrentPostType();
+		if ( postType !== 'post' ) {
+			return null;
+		}
+
+		const options = this.props.products.map( ( product ) => ( {
+			label: product,
+			value: product,
+		} ) );
+
+		// Free is always an option.
+		options.unshift( {
+			label: '— Free —',
+			value: 'openaccess',
+		} );
+
+		return (
+			<PluginDocumentSettingPanel
+				icon="money-alt"
+				initialOpen={ true }
+				title={ __( 'Reader revenue', 'google-site-kit' ) }
+			>
+				<PanelRow>
+					<SelectControl
+						label={ __( 'Access', 'google-site-kit' ) }
+						labelPosition="side"
+						onChange={ this.props.setAccess }
+						options={ options }
+						value={ this.props.access }
+					></SelectControl>
+				</PanelRow>
+				{ __( 'Preview this in the top admin bar', 'google-site-kit' ) }
+			</PluginDocumentSettingPanel>
+		);
+	}
+}
+
+export default compose(
+	withDispatch( ( dispatch ) => ( {
+		setAccess: ( access ) =>
+			dispatch( 'core/editor' ).editPost( {
+				meta: { sitekit__reader_revenue__access: access },
+			} ),
+	} ) ),
+	withSelect( ( select ) => ( {
+		access: select( 'core/editor' ).getEditedPostAttribute( 'meta' )
+			.sitekit__reader_revenue__access,
+		products: global.googlesitekit.data.select( STORE_NAME ).getProducts(),
```

Comment:
Hey @tofumatt ! Just wondering, do we have a good solution for using Site Kit's `withSelect` for a Gutenberg component? I tried to use Site Kit's `withSelect` alongside WordPress', in this call to `compose`, but then WordPress crashed and mentioned something about two versions of React. What do you think?

Reply:
cc @aaemnnosttv and @felixarntz

Reply:
>  I tried to use Site Kit's withSelect alongside WordPress', in this call to compose, but then WordPress crashed and mentioned something about two versions of React. What do you think?

Hey @ChrisAntaki, not sure if this is still an issue for you but the reason this is due to `withSelect` using hooks internally which requires that the same version of React is used and Site Kit's will always reference the version of React we bundle. So when using within the context of Gutenberg, we'll always need to use `withSelect` via the `wp` global for the same reason we need to use class-based components here.

If we don't already have an issue for using the two data stores together we'll need to create one for that.

Reply:
Thanks for the explanation, @aaemnnosttv !

Hey @felixarntz , just checking, did you already create an issue for this a while back ?

Reply:
@aaemnnosttv I created #4107 (and related #4108 + #4109) around this, but it doesn't actually encompass the ability to use stores from both registries in a single component yet I believe. Can you check where the new issue fits in with those?

Reply:
@felixarntz @ChrisAntaki I opened https://github.com/google/site-kit-wp/issues/4357. With this we would use the Site Kit registry for accessing both our stores and WP core stores – we would still need to use the HOCs via `wp` though as mentioned before.